src/BenchmarkDotNet/Engines/CacheClearingStrategies/AllocationsCacheClearingStrategy.cs

```diff
@@ -0,0 +1,166 @@
+                    var thread = new Thread(() => AllocateMemory(affinity)) { IsBackground = true, };
+                    thread.Start();
+
+                    threads[index++] = thread;
+                }
+            }
+
+            //Wait for all thread
+            foreach (var thread in threads)
+            {
+                thread.Join();
+            }
+        }
+
+        private void AllocateMemory(long affinity)
+        {
+            SetProcessorAffinity(affinity);
+
+            const int howManyMb = 6;
+            const int howManyPass = 10;
+            const int sizeOfOneArray = 64;
+            for (int i = 0; i < (howManyMb * 1024 * 1024 / sizeOfOneArray) * howManyPass; i++)
+            {
+                var tmpArray = new int[sizeOfOneArray];
+
+                for (int j = 0; j < sizeOfOneArray; j++)
+                {
+                    tmpArray[j] = i * j;
+                }
+
+                tmpArray[0] = 6;
+
+                for (int j = 0; j < sizeOfOneArray; j++)
+                {
+                    Consumer(tmpArray[j]);
+                }
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void Consumer(int v)
+        {
+
+        }
+
+        private static void SetProcessorAffinity(long coreMask)
+        {
+            int threadId = GetCurrentThreadId();
+            SafeThreadHandle handle = null;
+            var tempHandle = new object();
+            try
+            {
+                handle = OpenThread(0x60, false, threadId);
+                if (SetThreadAffinityMask(handle, new HandleRef(tempHandle, (IntPtr)coreMask)) == IntPtr.Zero)
+                {
+
+                    Console.WriteLine("Failed to set processor affinity for thread "+ Marshal.GetLastWin32Error());
+                }
+            }
+            finally
+            {
+                handle?.Close();
+            }
+        }
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+        private static extern IntPtr SetThreadAffinityMask(SafeThreadHandle handle, HandleRef mask);
+
+        [SuppressUnmanagedCodeSecurity]
+        private class SafeThreadHandle : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            public SafeThreadHandle() : base(true)
+            {
+
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                return CloseHandle(handle);
+            }
+        }
+
+        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.Success), DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true, ExactSpelling = true)]
+        private static extern bool CloseHandle(IntPtr handle);
+
+        [DllImport("kernel32")]
+        private static extern int GetCurrentThreadId();
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+        private static extern SafeThreadHandle OpenThread(int access, bool inherit, int threadId);
+    }
+}
```

Comment:
could you please:

1. move `extern`s to the top of the file to make it clear that this type is using WinAPI
2. order the private methods by the importance? I would like to see the methods that do the most important things at the top, the helpers like `GetAffinitiesForSelectedProcessors` at the bottom. Just to get good overview what this code is doing quickly
3. add `<summary>` comments to the methods that perform actual cache cleanup? just two sentences what these methods do and why
4. in the comments add a link to the website/book which describeds how to do it (I assume you have used one)